### ISSUE TITLE
Portable Linux builds

### ISSUE THREAD

**Issue by Ext3h** [rschurade/Ingnomia#107]
In the current state, interested developers can already build the game locally in their favorite distribution, and then play it, but the prebuilt binaries are not working yet.

At least one failure is caused by missing symbols around the introduction of sized-deallocations in recent gcc versions, which it incompatible with older c++ standard library implementations.

We need to find a set of compiler options, so that we can ensure our CI builds are actually portable.

**Comment by comaldave** [comment]
Dou we have a unit test script that verifies if it is working?

**Comment by Ext3h** [comment]
No, we don't. And neither do we have a test node available where we could just start the game with UI in lack of both XServer and OpenGL support. We can only test this on bare metal, in lack of any VM with OpenGL 4.3 support.

And to top it off, the problems we did hear about so far were all about libc conflicts on different distributions. Curiously, Qt was apparently easy enough to just swap out, and so far we didn't face any OS specific issues at runtime.

So the goal is just to find a configuration which is able to safely resolve all symbols on the most wide spread Linux distributions.

That means Ubuntu 18.04, 20.04, Arch, as well as the Steam Runtime.

If it can start up to visible UI, it should suffice.

**Comment by condac** [comment]
I use linux and some apps comes as an AppImage that seems to include all dependecies, AppImages never failed to work for me so far so the system seems to work ok for such cases. But I have no idea how hard it is to create and manage them, and one downside is that the resulting file might be quiet big download but that is easy price to pay to get all the right files included.
